bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/retriever/AbstractCategoriesRetriever.java

```diff
@@ -110,15 +131,18 @@ public void extendCategoryQueryWith(Consumer<CategoryTreeQuery> categoryQueryHoo
     protected String generateQuery(List<String> identifiers) {
         CategoryTreeQueryDefinition queryArgs = generateCategoryQuery();
         return Operations.query(query -> {
-            for (String identifier : identifiers) {
-                try {
-                    int value = Integer.parseInt(identifier);
-                    String alias = "category_" + identifier;
-                    query.withAlias(alias).category(q -> q.id(value), queryArgs);
-                } catch (NumberFormatException x) {
-                    // ignore
-                }
+            FilterEqualTypeInput identifiersFilter = new FilterEqualTypeInput().setIn(identifiers);
+            CategoryFilterInput filter;
+            if (UrlProvider.CategoryIdentifierType.UID.equals(identifierType)) {
+                filter = new CategoryFilterInput().setCategoryUid(identifiersFilter);
+            } else if (UrlProvider.CategoryIdentifierType.ID.equals(identifierType)) {
+                filter = new CategoryFilterInput().setIds(identifiersFilter);
+            } else {
+                throw new RuntimeException("Category identifier type is not supported");
```

Comment:
Instead of throwing an exception which might break the complete website, can you add a warning to the logs and just fall back to the ID query? Worst thing that can happen then is an empty category result list.